testdata/project-v2-addon/PROJECT

```diff
@@ -1,13 +1,32 @@
 domain: testproject.org
+layout: go.kubebuilder.io/v2,declarative.kubebuilder.io/v1
+projectName: project-v2-addon
 repo: sigs.k8s.io/kubebuilder/testdata/project-v2-addon
 resources:
-- group: crew
+- api:
+    crdVersion: v1beta1
+    namespaced: true
+  controller: true
+  domain: testproject.org
+  group: crew
   kind: Captain
+  path: sigs.k8s.io/kubebuilder/testdata/project-v2-addon/api/v1
   version: v1
-- group: crew
+- api:
+    crdVersion: v1beta1
+    namespaced: true
+  controller: true
+  domain: testproject.org
+  group: crew
   kind: FirstMate
+  path: sigs.k8s.io/kubebuilder/testdata/project-v2-addon/api/v1
   version: v1
-- group: crew
+- api:
+    crdVersion: v1beta1
+  controller: true
+  domain: testproject.org
+  group: crew
   kind: Admiral
+  path: sigs.k8s.io/kubebuilder/testdata/project-v2-addon/api/v1
   version: v1
-version: "2"
+version: "3"
```

Comment:
Why it changes the config? Should it really change it?

Reply:
Project config 2 doesn't have the layout field required for this change. It may not be needed when we allow per-command `--plugin` flags as we will be able to use `kb init --plugins="go/v2" ...` and add the declarative plugin in `kb create api` calls, not requiring the layout field in the project file any more.